subsys/bluetooth/audio/cap_commander.c

```diff
@@ -77,18 +81,253 @@ int bt_cap_commander_discover(struct bt_conn *conn)
+		LOG_DBG("param->count (%zu) is larger than CONFIG_BT_MAX_CONN (%d)", param->count,
+			CONFIG_BT_MAX_CONN);
+		return false;
+	}
+
+	CHECKIF(param->param == NULL) {
+		LOG_DBG("param->param is NULL");
+		return false;
+	}
+
+	for (size_t i = 0; i < param->count; i++) {
+		const struct bt_cap_commander_broadcast_reception_start_member_param *start_param =
+			&param->param[i];
+		const union bt_cap_set_member *member = &param->param[i].member;
+		const struct bt_cap_common_client *client =
+			bt_cap_common_get_client(param->type, member);
+
+		if (client == NULL) {
+			LOG_DBG("Invalid param->param[%zu].member", i);
+			return false;
+		}
+
+		CHECKIF(start_param->num_subgroups == 0) {
+			LOG_DBG("param->param[%d]->num_subgroups is 0", i);
+			return false;
+		}
+
+		CHECKIF(start_param->subgroups == NULL) {
+			LOG_DBG("param->param[%d]->subgroup is NULL", i);
+			return false;
+		}
+
+		for (size_t j = 0U; j < start_param->num_subgroups; j++) {
+			CHECKIF(start_param->subgroups[j].bis_sync == 0) {
+				LOG_DBG("param->param[%zu].subgroup[%zu].bis_sync is 0", i, j);
+			}
+		}
+
+		for (size_t j = 0U; j < i; j++) {
+			const union bt_cap_set_member *other = &param->param[j].member;
+
+			if (other == member) {
+				LOG_DBG("param->members[%zu] (%p) is duplicated by "
+					"param->members[%zu] (%p)",
+					j, other, i, member);
+				return false;
+			}
+		}
+	}
+
+	return true;
+}
+#endif /* CONFIG_BT_BAP_BROADCAST_ASSISTANT */
+
 int bt_cap_commander_broadcast_reception_start(
 	const struct bt_cap_commander_broadcast_reception_start_param *param)
 {
+#if defined(CONFIG_BT_BAP_BROADCAST_ASSISTANT)
+	struct bt_bap_broadcast_assistant_add_src_param add_src_param = {0};
+	const struct bt_cap_commander_proc_param *proc_param;
+	struct bt_cap_common_proc *active_proc;
+	struct bt_conn *conn;
+	int err;
+
+	if (bt_cap_common_proc_is_active()) {
+		LOG_DBG("A CAP procedure is already in progress");
+
+		return -EBUSY;
+	}
+
+	if (!valid_broadcast_reception_start_param(param)) {
+		return -EINVAL;
+	}
+
+	bt_cap_common_start_proc(BT_CAP_COMMON_PROC_TYPE_BROADCAST_RECEPTION_START, param->count);
+
+	broadcast_assistant_cb.add_src = cap_commander_ba_add_src_cb;
+	if (!ba_cb_registered && cap_commander_register_ba_cb() != 0) {
+		LOG_DBG("Failed to register VCP callbacks");
+
+		return -ENOEXEC;
+	}
+
+	active_proc = bt_cap_common_get_active_proc();
+
+	for (size_t i = 0U; i < param->count; i++) {
+		const struct bt_cap_commander_broadcast_reception_start_member_param *member_param =
+			&param->param[i];
+		struct bt_conn *member_conn =
+			bt_cap_common_get_member_conn(param->type, &member_param->member);
+
+		if (member_conn == NULL) {
+			LOG_DBG("Invalid param->members[%zu]", i);
+
+			return -EINVAL;
+		}
+
+		/* Store the necessary parameters as we cannot assume that the supplied parameters
+		 * are kept valid
+		 * TODO: consider putting this into a function
+		 */
+		active_proc->proc_param.commander[i].conn = member_conn;
```

Comment:
Suggest to store `active_proc->proc_param.commander[i]` in a local variable to make these lines shorter